src/geo/layer_definition.js

```diff
@@ -661,242 +456,169 @@ Map.prototype = {
-        for (var i = 0, len = token.length; i < len; i++) {
-          tokenParams.push("auth_token[]=" + token[i]);
-        }
-        url += "?" + tokenParams.join('&')
-      } else {
-        url += "?auth_token=" + token
-      }
-    }
-    return url;
-  },
-
-
-  setSQL: function(sql) {
-    throw new Error("SQL is read-only in NamedMaps");
-  },
-
-  setCartoCSS: function(sql) {
-    throw new Error("cartocss is read-only in NamedMaps");
-  },
-
-  getCartoCSS: function() {
-    throw new Error("cartocss can't be accessed in NamedMaps");
+  containTooltip: function() {
+    var layers =  this.options.layer_definition.layers;
+    for(var i = 0; i < layers.length; ++i) {
+      var tooltip = layers[i].tooltip;
+      if (tooltip && tooltip.fields && tooltip.fields.length) {
+        return true;
+      }
+    }
+    return false;
   },
 
-  getSQL: function() {
-    throw new Error("SQL can't be accessed in NamedMaps");
+  getSubLayer: function(index) {
+    var layer = this.layers[index];
+    layer.sub = layer.sub || new SubLayer(this, index);
+    return layer.sub;
   },
 
-  setLayer: function(layer, def) {
-    var not_allowed_attrs = {'sql': 1, 'cartocss': 1, 'interactivity': 1 };
+  getSubLayerCount: function() {
+    return this.getLayerCount();
+  },
 
-    for(var k in def.options) {
-      if (k in not_allowed_attrs) {
-        delete def.options[k];
-        throw new Error( k + " is read-only in NamedMaps");
-      }
+  getSubLayers: function() {
+    var layers = []
+    for (var i = 0; i < this.getSubLayerCount(); ++i) {
+      layers.push(this.getSubLayer(i))
     }
-    return Map.prototype.setLayer.call(this, layer, def);
-  },
+    return layers;
+  }
+};
 
-  removeLayer: function(layer) {
-    throw new Error("sublayers are read-only in Named Maps");
-  },
+function LayerDefinition(layerDefinition, options) {
+  MapBase.call(this, options);
+  this.endPoint = MapBase.BASE_URL;
+  this.setLayerDefinition(layerDefinition, { silent: true });
+}
 
-  createSubLayer: function(attrs, options) {
-    throw new Error("sublayers are read-only in Named Maps");
-  }, 
+/**
+ * Generates the MapConfig definition for a list of sublayers.
+ *
+ * ``sublayers`` should be an array, an exception is thrown otherwise.
+ *
+ */
+LayerDefinition.layerDefFromSubLayers = function(sublayers) {
 
-  addLayer: function(def, layer) {
-    throw new Error("sublayers are read-only in Named Maps");
-  },
+  if(!sublayers || sublayers.length === undefined) throw new Error("sublayers should be an array");
 
-  // for named maps the layers are always the same (i.e they are
-  // not removed to hide) so the number does not change
-  getLayerIndexByNumber: function(number) {
-    return +number;
-  }
+  var layer_definition = {
+    version: '1.0.0',
+    stat_tag: 'API',
+    layers: []
+  };
 
+  for (var i = 0; i < sublayers.length; ++i) {
+    layer_definition.layers.push({
+      type: 'cartodb',
+      options: sublayers[i]
```

Comment:
Hey @javisantana, I think there's something wrong here. We're adding adding everything in `sublayers[i]` to the `options` attribute. That's generating something like:
```
{
  type: 'cartodb',
  options: {
    sql: '...',
    cartocss: '...',
  }
}
```

But according to the spec, it should be something like:

```
{
  type: 'cartodb',
  sql: '...',
  cartocss: '...',
  options: {
    ...
  }
}
```

Also, methods like `Sublayer#setSQL` are setting those attribute inside of the `options` attribute, which seems wrong to me.

What do you think?